lib/membrane/rtp/vad.ex

```diff
@@ -130,87 +98,34 @@ defmodule Membrane.RTP.VAD do
     end
   end
 
-  defp handle_vad(buffer, rtp_timestamp, level, state) do
-    state = %{state | current_timestamp: rtp_timestamp}
-    state = filter_old_audio_levels(state)
-    state = add_new_audio_level(state, level)
-    audio_levels_vad = get_audio_levels_vad(state)
-    actions = [buffer: {:output, buffer}] ++ maybe_send_event(audio_levels_vad, state)
-    state = update_vad_state(audio_levels_vad, state)
-    {actions, state}
-  end
+  defp handle_vad(buffer, rtp_timestamp, level_in_dbov, state) do
+    level_in_db = 127 - level_in_dbov
+    updated_audio_levels = AudioLevelQueue.add(state.audio_levels, level_in_db)
 
-  defp filter_old_audio_levels(state) do
-    Enum.reduce_while(state.audio_levels, state, fn {level, timestamp}, state ->
-      if Ratio.sub(state.current_timestamp, timestamp)
-         |> Ratio.gt?(state.rtp_timestamp_increment) do
-        {_level, audio_levels} = Qex.pop(state.audio_levels)
-
-        state = %{
-          state
-          | audio_levels_sum: state.audio_levels_sum - level,
-            audio_levels_count: state.audio_levels_count - 1,
-            audio_levels: audio_levels
-        }
-
-        {:cont, state}
-      else
-        {:halt, state}
-      end
-    end)
-  end
```

Comment:
OK, you're right it handles out-of-order scenarios. It will still be broken for missing packets as you're not putting anything in the queue and it might happen that `IsSpeakingEstimator` will receive much more than the desired time interval since it's based on queue length.

I guess though unless it's trivial, it can be done in separate task/issue

Reply:
The previous algorithm didn't handle it. Only a `rtp_timestamp_increment` was measured (a quotient of time between packets and the clock_rate).
My idea for implementing it is:
  1. Measuring `rtp_timestamp_increment`.
  2. Computing the difference between new timestamp and previous timestamp.
  3. Checking how many timestamps there should be by dividing the difference by `rtp_timestamp_increment`.
  4. Padding the missing packets with an audio level set to `0`.

For example: if `rtp_timestamp_increment` is 960 and the difference between `current` and `previous` rtp timestamp is 1920, we add to the `audio_levels` queue:
  - one `0` for the missing packet and the current audio level value,
  - the value of audio level in the rtp packet that have arrived


This is a simple solution I have come up with. It solves one problem you have proposed. Is this enough?